lib/node-detector.js

```diff
@@ -0,0 +1,96 @@
+import {fs, cp} from './utils';
+import {default as log} from './logger';
+import path from 'path';
+
+const NODE_COMMON_PATHS = [
+  process.env.NODE_BIN,
+  '/usr/local/bin/node',
+  '/opt/local/bin/node',
+];
+
+class NodeDetector {
+  async retrieveInCommonPlaces () {
+    for(let p of NODE_COMMON_PATHS) {
+      if (p && await fs.exists(p)) {
+        log.info(`Node binary found at common place {p}.`);
+        return p;
+      }
+    }
+    log.info('Node binary wasn\'t found at common places.');
+    return null;
+  }
+
+  async retrieveUsingWhichCommand () {
+    let stdout;
+    try {
+      [stdout] = await cp.exec("which node", { maxBuffer: 524288});
+    } catch (ign) {}
+    let nodePath = stdout.replace("\n", "");
+    if(await fs.exists(nodePath)) {
+      log.info("Node binary found using which command at " + nodePath);
+      return nodePath;
+    } else {
+      log.info('Node binary not found using the which command.');
+      return null;
+    }
+  }
+
+  async retrieveUsingAppleScript () {
+    var appScript = [
+      'try'
+      , '  set appiumIsRunning to false'
+      , '  tell application "System Events"'
+      , '    set appiumIsRunning to name of every process contains "Appium"'
+      , '  end tell'
+      , '  if appiumIsRunning then'
+      , '    tell application "Appium" to return node path'
+      , '  end if'
+      , 'end try'
+      , 'return "NULL"'
+    ].join("\n");
+    let stdout;
+    try {
+      [stdout] = await cp.exec("osascript -e '" + appScript + "'", { maxBuffer: 524288});
+    } catch(ign) {}
+    let nodePath = stdout.replace("\n", "");
+    if(await fs.exists(nodePath)) {
+      log.info("Node binary found using AppleScript at " + nodePath);
+      return nodePath;
+    } else {
+      log.info('Node binary not found using AppleScript.');
+      return null;
+    }
+  }
+
+  async retrieveUsingAppiumConfigFile () {
```

Comment:
we're doing away with appiumconfig.json in appium 1.5, so I think we can get rid of this check